app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterMovementReferenceNumberController.scala

```diff
@@ -146,17 +135,11 @@ class EnterMovementReferenceNumberController @Inject() (
                 )
               ),
             movementReferenceNumber =>
-              movementReferenceNumber.value match {
-                case Left(entryNumber) =>
-                  val updatedAnswers = answers.fold(
-                    _ =>
-                      CompleteMovementReferenceNumberAnswer(
-                        Left(entryNumber)
-                      ),
-                    complete => complete.copy(movementReferenceNumber = Left(entryNumber))
-                  )
-                  val newDraftClaim  =
-                    fillingOutClaim.draftClaim.fold(_.copy(movementReferenceNumberAnswer = Some(updatedAnswers)))
+              movementReferenceNumber match {
+                case entryMovementReferenceNumber @ MovementReferenceNumber(Left(_))  =>
+                  val newDraftClaim =
+                    fillingOutClaim.draftClaim
+                      .fold(_.copy(movementReferenceNumberAnswer = Option(entryMovementReferenceNumber)))
```

Comment:
once again, we lose one fold by getting rid of the inner option which never happens.